crates/store/src/state.rs

```diff
@@ -888,6 +888,11 @@ impl State {
         self.db.select_network_account_by_prefix(id_prefix).await
     }
 
+    /// Returns account IDs for all public (on-chain) network accounts.
+    pub async fn get_all_network_accounts(&self) -> Result<Vec<AccountId>, DatabaseError> {
+        self.db.select_all_network_account_ids().await
+    }
```

Comment:
This is fine for now, but in the future this would need to be paginated so that we don't return too many account IDs in a single response. Let's keep track of this as a follow-up issue.